bundler/helpers/v2/lib/functions/lockfile_updater.rb

```diff
@@ -36,7 +37,7 @@ def generate_lockfile
 
         old_reqs.each do |dep_name, old_req|
           d_dep = definition.dependencies.find { |d| d.name == dep_name }
-          if old_req == :none then definition.dependencies.delete(d_dep)
+          if old_req.to_s == DEPENDENCY_DROPPED then definition.dependencies.delete(d_dep)
```

Comment:
This started to blow up as the new gem requirement sorting tried to call `_requirements_sorted?` on `:none` https://github.com/rubygems/rubygems/blob/32ac0ba0cdad8ffe522865196d1a2ec3bbf62e08/bundler/lib/bundler/rubygems_ext.rb#L137